src/main/scala/com/picnicml/doddlemodel/clustering/RandomizableClusterer.scala

```diff
@@ -0,0 +1,26 @@
+package com.picnicml.doddlemodel.clustering
+
+import java.io.Serializable
+
+import breeze.linalg.DenseVector
+import com.picnicml.doddlemodel.base.Clusterer
+import com.picnicml.doddlemodel.data.{Features, Target}
+
+import scala.util.Random
+
+trait RandomizableClusterer[A <: RandomizableClusterer[A]] extends Clusterer[A] {
```

Comment:
As already mentioned, `fit` here shouldn't be protected. On the other hand that introduces another problem: API (public functions defined in abstract classes and traits) should be in the `base` package. If you take a look at the `linear.LinearModel` and other traits there, none of them define any public interface, they only encapsulate functionality that is common to all linear models. We should identify what is common to all clustering estimators and expose that in the form of a base class(es).